### ISSUE TITLE
Async Components

### ISSUE THREAD

**Issue by avalero** [denoland/fresh#2336]
Is there any way to define async components? It would be nice to be able to delegate server side data fetching to components.

**Comment by marvinhagemeister** [comment]
No, not at the moment. The closest we have is async routes, see https://fresh.deno.dev/docs/concepts/routes#async-route-components . Support for true async components in Preact on the server got merged a few days ago, and we'll likely adopt that model sometime in the future. There is no ETA for that yet though.

**Comment by vicary** [comment]
@marvinhagemeister that was wonderful! Albeit no ETA, thanks for making true server components a reality.

**Comment by avalero** [comment]
Thanks!

**Comment by marvinhagemeister** [comment]
Re-opening until we've landed truly async components.

**Comment by Hajime-san** [comment]
I think that the approach to async of Solid is a right way.
https://github.com/solidjs/solid/releases/tag/v2.0.0-beta.0

Throwing a `Promise` on the server could be really bottleneck.
https://x.com/hichaelmart/status/1979311874140442950
